build-support/bin/prune_travis_cache.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash -e
+
+function nuke_if_too_big() {
+  path=$1
+  limit_mb=$2
+  echo "## Pruning ${path} (if larger than ${limit_mb}MB)..."
+  if [ -d ${path} ]; then
+    actual_mb=$(du -m -d0 "${path}" | cut -f 1)
+    echo "Size of ${path}: ${actual_mb}MB"
+    if (( actual_mb > limit_mb )); then
+      echo "${path} is too large, nuking it."
+      rm -rf "${path}"
+    else
+      echo "${path} is not too large, leaving it."
+    fi
+  else
+    echo "Directory ${path} doesn't exist."
+  fi
+}
+
+
+# Prune the dirs we cache on travis if they get too big.
+# Note that some of these dirs are only relevant to native code building shards, and others only to
+# pants running shards. And some only to linux or only to osx.
+# However there's no harm in checking them all in all cases.
+
+nuke_if_too_big "${HOME}/.pants_pyenv" 512
+nuke_if_too_big "${HOME}/.aws_cli" 128
+nuke_if_too_big "${HOME}/.cache/pants/lmdb_store" 1024
+nuke_if_too_big "${HOME}/.cache/pants/tools" 128
+nuke_if_too_big "${HOME}/.cache/pants/zinc" 128
+nuke_if_too_big "${HOME}/.ivy2/pants" 512
+nuke_if_too_big "${HOME}/.npm" 128
+nuke_if_too_big "${HOME}/.cache/pants/rust/cargo" 1024
```

Comment:
We probably want to unconditionally keep the `git` subdir of this directory (because a submodule checkout of grpc-rs takes _forever_). Ideally we'd also keep `.crates.toml` and `bin` because they contain a release-compiled rust crate which is a little slow to install.

Reply:
afaict almost the entire size of `.cache/pants/rust/cargo` is actually just the grpc checkout's `.git` and those of its submodules `boringssl` and `boringssl-with-bazel`.  What is going on in there?? Is there a way to get rust to do a shallow clone?

Reply:
Meanwhile, made this trim just the /registry subdir.

Reply:
Sadly we need the submodules :(

Hopefully at some point we can just migrate to Tonic, and remove this dep entirely.

Reply:
By shallow I meant without the git history, which is almost the entire size of this.